istioctl/cmd/analyze.go

```diff
@@ -308,15 +309,26 @@ istioctl experimental analyze -L
 	return analysisCmd
 }
 
-func gatherFiles(args []string) ([]string, error) {
-	var result []string
-	for _, a := range args {
-		if _, err := os.Stat(a); err != nil {
-			return nil, fmt.Errorf("could not find file %q", a)
+func gatherFiles(args []string) ([]io.Reader, error) {
+	var readers []io.Reader
+	var r *os.File
+	var err error
+	for _, f := range args {
+		if f == "-" {
+			if isatty.IsTerminal(os.Stdin.Fd()) {
```

Comment:
Should this still be here? Why do we need a tty to handle reading from stdin?

I admit I'm not that familiar with when ttys are used or not, but rather than failing in this case (or occasionally treating it as a filename), why not just always make '-' refer to stdin?

Reply:
I admit that I was thinking that piping/redirecting was the only use case, but I guess if we're passing in hyphen then we might as well block on terminal input until they ctrl+d. Good point.